lib/puppet/parser/scope.rb

```diff
@@ -587,7 +587,7 @@ def define_settings(type, params)
     }
   end
 
-  RESERVED_VARIABLE_NAMES = ['trusted', 'facts'].freeze
+  RESERVED_VARIABLE_NAMES = ['trusted', 'facts', 'server_facts'].freeze
```

Comment:
Should server_facts be a reserved name when the :trusted_server_facts setting is true?